Update IDP configs to have self service SSO (#2081)
astronomer/docs: astro/configure-idp.md

#### Step 1: Create a SAML-based connection to Okta

To set up Okta as your IdP, you will create a Security Assertion Markup Language (SAML) connection to Okta.

1. In the Cloud UI, click **Settings**, then click **Authentication**.
2. In the **SSO Configuration** menu, click **Configure SSO**.
3. Configure the following values for your connection:

    - **Connection type**: Select **SAML**.
    - **SSO Domain(s)**: Enter the verified domain(s) that you want to map to Okta.
    - **Automatic Membership**: Set the default role for users who join your Organization through Okta and without an explicit invite from Astro.

4. Copy the **Single Sign On URL** and **Audience URI (SP ENTITY ID)** for the next step.
5. Open a new tab and go to Okta. In the Okta Admin Console, create a SAML app integration. See [Create SAML app integrations using AIW](https://help.okta.com/en/prod/Content/Topics/Apps/Apps_App_Integration_Wizard_SAML.htm). Complete the following fields:

    - **App logo**: (Optional) Use the [official Astronomer monogram asset](https://github.com/astronomer/docs/tree/main/static/img/assets/astronomer-monogram-rgb-600px.png) hosted on GitHub.

>     - **Single sign on URL**: `<your-sso-url>`

Author: Yea I think this is how it is in their UI, which is usually what we defer to in terms of style

#### Prerequisites

- A [Ping Identity account](https://www.pingidentity.com/) with administrative access.
- [Organization Owner](user-permissions.md) privileges in the Organization you're configuring.
- At least one [verified domain](manage-domains.md).

#### Step 1: Configure Ping Identity

1. In the Cloud UI, click **Settings**, then click **Authentication**.
   
2. In the **SSO Configuration** menu, click **Configure SSO**.
   
3. Configure the following values for your connection:

    - **Connection type**: Select **SAML**.
    - **SSO Domain(s)**: Enter the verified domain(s) that you want to map to Ping Identity.
    - **Automatic Membership**: Set the default role for users who join your Organization through Ping Identity and without an explicit invite from Astro.

> 4. Copy the **Single Sign On URL** and **Audience URI (SP ENTITY ID)** for Step 8.

Reviewer: ```suggestion
4. Copy the **Single Sign-On URL** and **Audience URI (SP ENTITY ID)** for Step 8.
```

5. Open a new tab and go to PingIdentity. In the PingIdentity Administrator Console, click **Connections** in the left pane, and then click the **+** icon next to **Applications**.

6. In the **Application Name** field, enter `Astro`. Optionally, add a description and an icon.

7. Click **SAML Application**, and then click **Configure**.

8. Click **Manually Enter** and then complete the following fields:

    - **ACS URLs**: `<your-sso-url>`
    - **Entity ID**: `<your-audience-uri>`

9.  Click **Save**.

> 10. Click **Edit** on the **Overview** page, and then enter `<your-sso-url>` in the **Signon URL** field. Then, click **Save**.

Reviewer: ```suggestion
10. Click **Edit** on the **Overview** page, and then enter `<your-sso-url>` in the **Sign on URL** field. Then, click **Save**.
```

11. Click the **Configuration** tab, and then click **Edit**.

12. Select **Sign Assertion & Response** and confirm `RSA_SHA256` is selected in the **Signing Algorithm** list. Then, click **Save**.

13. On the **Configuration** page, click **Download Signing Certificate** and select `X509 PEM (.crt)` to download the X.509 certificate for your application.

> 14. Copy and save the URL in the **Single Signon Service** field.

Reviewer: ```suggestion
14. Copy and save the URL in the **Single Sign-on Service** field.
```

- [Organization Owner](user-permissions.md) privileges in the Organization you're configuring.
- A list of domains that you own and should be authorized to access your Astro Organization.

#### Step 1: Verify your domain(s)

Mapping a domain to Ping Identity ensures that all users with the same email address domain have the same authentication experience when they log in to Astro. You must map at least one domain to Ping Identity to complete this setup. You can later use this mapping to enforce specific login methods for users with emails from a specific domain. 

To map a domain to Ping Identity, you must verify that you own the domain.

1. In the Cloud UI, click **Settings**, then click **Authentication**.
2. In the **Managed Domains** menu, click **Managed Domain**.
3. In the **Domain** field, enter the domain that you want to map to Ping Identity.
4. Click **Create**. The domain is added to your **Managed Domains** and marked as **Unverified**.
5. In the entry for your domain, click **Verify**.
6. Follow the steps provided by the Cloud UI to verify your domain. 
7. Repeat steps 1-6 for any other domains you want to map to Ping Identity.

#### Step 2: Configure Ping Identity

> 1. In the Cloud UI, click **Settings**, then click **Authentication**.

Author: This setup takes sooooo many steps, I'd love to find a way to break it out into topics or reduce the step count. I didn't want to break it into separate topics because all of this configuration takes place in a single "scenario", which in this case configuring the connection from Astro.

I'd also be up for outsourcing some of the Ping/ OneLogin steps to their own documentation.

    | lastName          | Family Name     |
    | name              | Formatted       |

16. Click **Save**.

17. Click the toggle in the top right to enable the application.
    
18. Assign yourself to Astro from Ping Identity. See [Editing a user](https://docs.pingidentity.com/r/en-us/pingone/p1_t_edituser?section=gnn1564020489010).
    
19. Return to the Cloud UI. In the configuration screen for your SAML connection, configure the following values:

>     - **Identity Provider Single Sign-on URL**: Enter the value you copied from the **Single Signon Service** field.

Reviewer: ```suggestion
    - **Identity Provider Single Sign-on URL**: Enter the value you copied from the **Single Sign-on Service** field.
```

please tell me their UI doesn't say *signon* :(

Author: Ok just checked again and THANKFULLY no.. this must have been left over from the Notion draft

:::

An SSO bypass link allows you to authenticate to your Organization without using SSO. This link should be used to access your Organization only when you can't access Astro due to an issue your identity provider.

1. In the Cloud UI, click the **Settings** tab.
   
2. In the **SSO Bypass Link** field, click **Copy**. Save this link for when you need to log in to Astro without using SSO.

If you don't want to maintain an SSO bypass link, click **Delete**. You can always regenerate a link if you need one in the future. 

#### Step 3: Assign users to your Ping Identity application

Assign users from your organization to your new application. See [Managing user groups](https://docs.pingidentity.com/bundle/pingcentral-19/page/awu1616596133840.html).

When a user assigned to the application accesses Astro, they are automatically signed in to Ping Identity after entering their email in the Cloud UI.

</TabItem>

</Tabs>

## Advanced setup

### Disable just-in-time provisioning

> Astro supports just-in-time provisioning by default for all [single sign-on (SSO) integrations](configure-idp.md). This means that if someone without an Astronomer account tries logging into Astronomer with an email address from a domain that you manage, they are automatically granted a default role in your Organization without needing an invite. Users with emails outside of this domain need to be invited to your Organization to access it. 

Reviewer: Does that link take users just to this page? Should we instead have a link to one of the initial headers?

Author: good catch, this was leftover from when these docs were on a separate page

Contact [Astronomer support](https://cloud.astronomer.io/support) for assistance with enabling or disabling just-in-time provisioning.

### Restrict authentication options 

By default, users have access to all possible authentication methods when logging into Astro. You can remove specific authentication options, such as GitHub or Google, so that users can use only the methods that your Organization wants to support, such as your third party identity provider.

> To restrict which authentication options are available on Astro for your organization, contact [Astronomer support](https://cloud.astronomer.io/support).

Reviewer: ```suggestion
To restrict available authentication options on Astro for your organization, contact [Astronomer support](https://cloud.astronomer.io/support).
```

If you don't want to maintain an SSO bypass link, click **Delete**. You can always regenerate a link if you need one in the future. 

#### Step 3: Assign users to your Ping Identity application

Assign users from your organization to your new application. See [Managing user groups](https://docs.pingidentity.com/bundle/pingcentral-19/page/awu1616596133840.html).

When a user assigned to the application accesses Astro, they are automatically signed in to Ping Identity after entering their email in the Cloud UI.

</TabItem>

</Tabs>

## Advanced setup

### Disable just-in-time provisioning

Astro supports just-in-time provisioning by default for all single sign-on (SSO) integrations. This means that if someone without an Astronomer account tries logging into Astronomer with an email address from a domain that you manage, they are automatically granted a default role in your Organization without needing an invite. Users with emails outside of this domain need to be invited to your Organization to access it. 

Contact [Astronomer support](https://cloud.astronomer.io/support) for assistance with enabling or disabling just-in-time provisioning.

### Restrict authentication options 

> By default, users can choose any of the [available authentication methods](#SSO-authorization-identity-providers) when logging into Astro. You can remove specific authentication options, such as GitHub or Google, so that users can use only the methods that your Organization wants to support, like your third party identity provider.

Author: ```suggestion
By default, users can choose any of the [available authentication methods](#supported-sso-identity-providers) when logging into Astro. You can remove specific authentication options, such as GitHub or Google, so that users can use only the methods that your Organization wants to support, like your third party identity provider.
```

:::

An SSO bypass link allows you to authenticate to your Organization without using SSO. This link should be used to access your Organization only when you can't access Astro due to an issue your identity provider.

1. In the Cloud UI, click the **Settings** tab.
   
2. In the **SSO Bypass Link** field, click **Copy**. Save this link for when you need to log in to Astro without using SSO.

If you don't want to maintain an SSO bypass link, click **Delete**. You can always regenerate a link if you need one in the future. 

#### Step 3: Assign users to your Ping Identity application

Assign users from your organization to your new application. See [Managing user groups](https://docs.pingidentity.com/bundle/pingcentral-19/page/awu1616596133840.html).

When a user assigned to the application accesses Astro, they are automatically signed in to Ping Identity after entering their email in the Cloud UI.

</TabItem>

</Tabs>

## Advanced setup

### Disable just-in-time provisioning

> Astro supports just-in-time provisioning by default for all [single sign-on (SSO) integrations](configure-idp.md). This means that if someone without an Astronomer account tries logging into Astronomer with an email address from a domain that you manage, they are automatically granted a default role in your Organization without needing an invite. Users with emails outside of this domain need to be invited to your Organization to access it. 

Author: ```suggestion
Astro supports just-in-time provisioning by default for all single sign-on (SSO) integrations. This means that if someone without an Astronomer account tries logging into Astronomer with an email address from a domain that you manage, they are automatically granted a default role in your Organization without needing an invite. Users with emails outside of this domain need to be invited to your Organization to access it. 
```

    
21. Return to the Cloud UI. In the configuration screen for your SAML connection, configure the following values:

    - **Identity Provider Single Sign-on URL**: Enter the value you copied from the **Single Signon Service** field.
    - **X.509 Certificate**: Enter the X.509 Certificate that you downloaded.

22. Click **Create**. Your Ping Identity integration appears as an entry in **SSO Configuration**.
    
23. In **SSO Configuration**, click **Activate**. You are redirected to Ping Identity to test your configuration. After you have successfully authenticated, you are redirected to Astro.
    
24. Click **Activate SSO**.
    
25. Copy the provided **SSO bypass link** and store it somewhere safe. See [Bypass single sign-on](manage-organization.md#bypass-single-sign-on).

#### Step 3: Assign users to your Ping Identity application

Assign users from your organization to your new application. See [Managing user groups](https://docs.pingidentity.com/bundle/pingcentral-19/page/awu1616596133840.html).

When a user assigned to the application accesses Astro, they are automatically signed in to Ping Identity after entering their email in the Cloud UI.

</TabItem>

</Tabs>

> ## Frequently Asked Questions

Author: I'm inclined to keep this out, because a user would learn the answer to these questions as they're configuring the IDP themselves based on the responses from Astro. How do these questions help them make a decision?
